### DISCUSSION TITLE
UK 240v 50hz

### DISCUSSION THREAD

**Discussion by frogsuk** [visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA#277] [Q&A] [answered]
Background:
I have a Helsinki 2000 with Egg shape. The heater is the later rectangular plastic housing and NOT the Aluminium housing. It has frost control and is six wired controller. 
Question 1.:
Would this wonderful circuit work on this type of egg?
Question 2.: 
I have a 3.9kw solar array on my roof and an additional 14.7kw LifePo4 DIY battery set up with a Sofar ME3000SE battery charger/inverter. I would like to automate the hot tub to come on when there is suitable solar power and also battery power. The tub is mainly used at approximately 10am for 1 hour every day so my wife can do her exercises as the hospital has closed the hydrotherapy pool since covid. She has chronic rheumatoid arthritis. I’m also in a wheelchair and maintain the tub for her. This PCB would be an absolute Godsend for us both to a.) save a bit of £ heating it and b.) make my life easier in controlling it.
To this end, do you think I am living in cloud cuckoo land thinking that I may be able to achieve the above as well?
Thanks all in advance.
(early retired engineer)

**Comment by visualapproach** [comment, marked as answer]
Hi @frogsuk! I'm touched by your situation and to be part of making your life easier. I'm not sure what aluminum pump you're referring to but your goal sounds doable! Most of the 6-wire pumps is compatible, although some models require extra persuation to work. As is, the device can be scheduled to turn on or off thing by time. With some additional coding it can be reacting to inputs on the spare gpio pins. If you have a home automation system it adds a lot of possibilities.
I have a couple of crazy long work days ahead but please post some pictures of the pump or if you can see the model number. It's usually on the sticker on the pump. And do you have any means to get info about the battery/solar states to the esp?

**Reply by frogsuk** [reply]
Sorry, not the pump. I meant to say the heater body is plastic rectangular and not aluminium rectangular. 
I have the pump on my brain at the moment and am a bit peeved at the poor design as I have just had to replace the seals on mine this evening due to water gushing out. The housing body was warped, the screws loose and the rubber seal perished.
Fortunately, I have an old flat engineer's surface plate here, so I spent 30 minutes with some 180 grit paper and the surfaces are no longer warped. I also replaced the old seal with a red Viton o ring, with a gasket compound to take up all the rest of the poor design. This will be the tub's 3rd season.
So, it's the brown egg-shaped unit for the Helsinki tub and has a plastic housed heat exchanger and not aluminium. 
I'm very sorry for the confusion but thank you so much for the time to give that full in depth answer.

**Comment by visualapproach** [comment]
No problem! I agree, they don't hold forever those things. I just threw my liner away due to unfixable deflation after a couple of years use. Made some DIY silicone seals, changed impeller and liner over the last years, and plugging the air back valve.
Just post here if you need some more advice with this project. Hope I can help.
